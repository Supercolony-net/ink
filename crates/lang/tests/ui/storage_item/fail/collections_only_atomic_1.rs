use ink_prelude::vec::Vec;
use ink_storage::StorageValue;

#[ink_lang::storage_item]
struct NonAtomic {
    a: StorageValue<u128>,
}

#[ink_lang::storage_item]
struct Contract {
    a: Vec<NonAtomic>,
}

fn main() {}
