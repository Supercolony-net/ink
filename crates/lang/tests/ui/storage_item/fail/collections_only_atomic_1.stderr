error[E0277]: the trait bound `NonAtomic: AtomicGuard<true>` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:9:1
   |
9  | #[ink_lang::storage_item]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AtomicGuard<true>` is not implemented for `NonAtomic`
   |
   = help: the trait `AtomicGuard<false>` is implemented for `NonAtomic`
   = note: required because of the requirements on the impl of `StorageType<()>` for `Vec<NonAtomic>`
   = note: required because of the requirements on the impl of `AutoStorageType<ManualKey<2085512762_u32>>` for `Vec<NonAtomic>`
note: required because it appears within the type `Contract`
  --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^
note: required by a bound in `EncodeLike`
  --> $CARGO/parity-scale-codec-3.1.2/src/encode_like.rs
   |
   | pub trait EncodeLike<T: Encode = Self>: Sized + Encode {}
   |                                         ^^^^^ required by this bound in `EncodeLike`
   = note: this error originates in the derive macro `::scale::Encode` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NonAtomic: AtomicGuard<true>` is not satisfied
   --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:9:1
    |
9   | #[ink_lang::storage_item]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AtomicGuard<true>` is not implemented for `NonAtomic`
    |
    = help: the trait `AtomicGuard<false>` is implemented for `NonAtomic`
    = note: required because of the requirements on the impl of `StorageType<()>` for `Vec<NonAtomic>`
    = note: required because of the requirements on the impl of `AutoStorageType<ManualKey<2085512762_u32>>` for `Vec<NonAtomic>`
note: required because it appears within the type `Contract`
   --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:10:8
    |
10  | struct Contract {
    |        ^^^^^^^^
note: required by a bound in `Decode`
   --> $CARGO/parity-scale-codec-3.1.2/src/codec.rs
    |
    | pub trait Decode: Sized {
    |                   ^^^^^ required by this bound in `Decode`
    = note: this error originates in the derive macro `::scale::Decode` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NonAtomic: AtomicGuard<true>` is not satisfied
   --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:9:1
    |
9   | #[ink_lang::storage_item]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AtomicGuard<true>` is not implemented for `NonAtomic`
    |
    = help: the trait `AtomicGuard<false>` is implemented for `NonAtomic`
    = note: required because of the requirements on the impl of `StorageType<()>` for `Vec<NonAtomic>`
    = note: required because of the requirements on the impl of `AutoStorageType<ManualKey<2085512762_u32>>` for `Vec<NonAtomic>`
note: required because it appears within the type `Contract`
   --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:10:8
    |
10  | struct Contract {
    |        ^^^^^^^^
note: required by a bound in `Result`
   --> $RUST/core/src/result.rs
    |
    | pub enum Result<T, E> {
    |                 ^ required by this bound in `Result`
    = note: this error originates in the derive macro `::scale::Decode` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NonAtomic: AtomicGuard<true>` is not satisfied
 --> tests/ui/storage_item/fail/collections_only_atomic_1.rs:9:1
  |
9 | #[ink_lang::storage_item]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AtomicGuard<true>` is not implemented for `NonAtomic`
  |
  = help: the trait `AtomicGuard<false>` is implemented for `NonAtomic`
  = note: required because of the requirements on the impl of `StorageType<()>` for `Vec<NonAtomic>`
  = note: required because of the requirements on the impl of `AutoStorageType<ManualKey<2085512762_u32>>` for `Vec<NonAtomic>`
  = note: this error originates in the attribute macro `ink_lang::storage_item` (in Nightly builds, run with -Z macro-backtrace for more info)
